tests/pipeline/test_filter.py

```diff
@@ -379,3 +382,37 @@ def test_isfinite(self):
             initial_workspace={self.f: data},
         )
         check_arrays(results['isfinite'], isfinite(data))
+    
+    def test_window_safe(self):
+        # all true data set of (days, securities)
+        data = full(self.default_shape, True, dtype=bool)
+
+        # rolling window length for TestFactor
+        k = 8
+
+        class InputFilter(Filter):
+            inputs = ()
+            window_length = 0
+
+        class TestFactor(CustomFactor):
+            dtype = float64_dtype
+            inputs = (InputFilter(), )
+            window_length = k
+
+            def compute(self, today, assets, out, filter_):
+                # sum for each column
+                out[:] = nansum(filter_, axis=0)
```

Comment:
[NaN is only a thing for floating-point values](https://en.wikipedia.org/wiki/IEEE_floating_point), so using `nansum` doesn't really make sense here since your inputs are bools.  (It's harmless, since it should just fall back to `sum`, but it's a little misleading to the reader.).  I would just use `filter_.sum`.